game-engine/src/main/scala/za/co/entelect/challenge/game/contracts/map/CarGameMap.scala

```diff
@@ -158,13 +154,28 @@ class CarGameMap(players: util.List[Player], mapGenerationSeed: Int, lanes: Int,
     val endBlockNumber = endPosition.getBlockNumber();
     val blocksWithObject =
       blocks.find(x =>
-        (x.getPosition().getLane() == endLane && x.getPosition().getBlockNumber() >= startBlockNumber && x.getPosition().getBlockNumber() <= endBlockNumber) &&
+        (x.getPosition().getLane() == endLane && x.getPosition().getBlockNumber() >= startBlockNumber && x.getPosition().getBlockNumber() <= endBlockNumber)
+            &&
           x.getMapObject() == mapObject
       );
     val pathIncludesObject = blocksWithObject.isDefined;
     return pathIncludesObject;
   }
 
+  private def numberOfMapObjectsInPath(startPosition: BlockPosition, endPosition: BlockPosition, mapObject: Int): Int = {
+    val startLane = startPosition.getLane();
+    val startBlockNumber = startPosition.getBlockNumber();
+    val endLane = endPosition.getLane();
+    val endBlockNumber = endPosition.getBlockNumber();
+    val blocksWithObject =
+      blocks.count(x =>
+        (x.getPosition().getLane() == endLane && x.getPosition().getBlockNumber() >= startBlockNumber && x.getPosition().getBlockNumber() <= endBlockNumber)
+            &&
+            x.getMapObject() == mapObject
```

Comment:
The formatting for this part looks a bit funky
```suggestion
    val blocksWithObject = blocks.count(b => (b.getPosition().getLane() == endLane 
                                               && b.getPosition().getBlockNumber() >= startBlockNumber 
                                               && b.getPosition().getBlockNumber() <= endBlockNumber)
                                           && b.getMapObject() == mapObject
```